WordPress/gutenberg
# Prepare `WP_Theme_JSON_Data` class for backporting

## What

This PR follows-up https://github.com/WordPress/gutenberg/pull/44015 addressing some post-merge feedback for improving code quality and make it easier to backport.

## How

- Class renamed from `WP_Theme_JSON_Data` to `WP_Theme_JSON_Data_Gutenberg`.
- Document the new filters inline.
- Create specific temp variables for the filters (input/output) and don't overload the existing.

## How to test

Make sure it still works as before.

Use the filters to modify some of the data. For example, this is how the theme data can be modified:

```php
<?php

function filter_global_styles_theme( $theme_json ){
        $new_data = array(
                'version'  => 2,
                'settings' => array(
                        'color' => array(
                                'text'  => false, /* disable text color UI */
                                'palette'    => array( /* New palette */
                                        array(
                                                'slug'  => 'foreground',
                                                'color' => 'red',
                                                'name'  => 'Foreground',
                                        ),
                                ),
                        ),
                ),
        );
        return $theme_json->update_with( $new_data );
}
add_filter( 'global_styles_theme', 'filter_global_styles_theme' );
```
